tss-esapi/src/structures/tagged/schemes.rs

```diff
@@ -543,3 +543,114 @@ impl TryFrom<RsaScheme> for RsaDecryptionScheme {
         }
     }
 }
+
+/// Full description of signature schemes.
+///
+/// # Details
+/// Corresponds to `TPMT_SIG_SCHEME`.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum SignatureScheme {
+    RsaSsa { hash_scheme: HashScheme },
+    RsaPss { hash_scheme: HashScheme },
+    EcDsa { hash_scheme: HashScheme },
+    Sm2 { hash_scheme: HashScheme },
+    EcSchnorr { hash_scheme: HashScheme },
+    EcDaa { ecdaa_scheme: EcDaaScheme },
+    Hmac { hmac_scheme: HmacScheme },
+    Null { hash_scheme: HashScheme },
```

Comment:
is the inner `hash_scheme` needed here? based on the spec it seems when the selector is null the details are irrelevant

Reply:
hrrm I think it is used because of 'any'. But I think it would be better to add Any instead and let Null be Null. Or what do you think?

Reply:
I'm not entirely sure, the spec isn't very clear - what exact selector corresponds to `Any`? I don't know if I saw one... I did see that for `Null` you don't have any inner type

Reply:
Any is used to set hash_scheme for any of the selectors that has a hash scheme. So it would look exactly as Null does today.

Reply:
Ah, fair, yeah, then rebranding this one as `Any` and having a separate, empty `Null` sounds good

Reply:
Hrrmm it was not that easy. There is no way to parse 'Any' it has no selector. So I think I just will skip it.

Reply:
I have been looking through every specification I could find in order to figure out how any works. And I have still not really understood it. So I need to do some more digging before fixing this.

Reply:
Ok I think I have come up with suitable substitute for ```anySig```. Don't know if it something we want though.

Reply:
Is that the new `set_signing_scheme` method below? I'm ok with that as an option. Still think `Null` should be void of parameters, though

Reply:
That is just me not paying attention. It should be void.